crucible-mir/src/Mir/Trans.hs

```diff
@@ -1315,7 +1395,59 @@ evalPlaceProj ty pl@(MirPlace tpr ref NoMeta) M.Deref = do
         let dynRefVtable = S.getStruct dynRefVtableIndex dynRef
         return $ MirPlace C.AnyRepr dynRefData (DynMeta dynRefVtable)
 
-evalPlaceProj ty pl@(MirPlace tpr ref NoMeta) (M.PField idx _mirTy) = do
+    -- For (dynamically-sized) ADTs wrapping slices, with MirPlace input of the
+    -- form:
+    --
+    --   MirPlace (*S<[T]>) <expr: **S<[T; len]> NoMeta
+    --
+    -- We produce output of the form:
+    --
+    --   MirPlace S<[T]> <expr: *S<[T; len]> (SliceMeta len)
+    --
+    -- Where `len` is the wrapped slice's length.
+    doSliceAdt :: M.Ty -> R.Expr MIR s MirReferenceType -> MirGenerator h s ret (MirPlace s)
+    doSliceAdt adtTy ref' =
+      do
+        -- Normally we'd use `tyToReprM adtTy` to get the ADT's representation,
+        -- but that would involve computing its slice field's representation,
+        -- which should error
+        let adtRepr = R.exprType ref'
```

Comment:
If `R.exprType` gives the `CrucibleType` of an `R.Expr`, then this will always return `MirReferenceType`, which is definitely not the repr of the ADT.  But (as mentioned above) the `MirPlace` `TypeRepr` doesn't make a lot of sense here, so maybe we can get away with using `AnyType` or something as a placeholder.